### ISSUE TITLE
List Filter Control Label - Provide a valid label for form fields - (2036570437)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#5626]
### Module: 
~39 Value List and Value List Item~
List and List Item

### Violation: 
 Provide a valid label for the component's filter list item

![image](https://user-images.githubusercontent.com/5023024/202567795-5b898332-e277-4076-840c-22b1607ed40e.png)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570437 

### Severity: 
 10 

### Description: 
 [Issue]
There is filter form control in the "Test 2. Value List with Filter" section without a programmatically associated label describing their purpose.

[User Impact]
Screen reader users and users with cognitive disabilities may be unable to determine what these controls are for. Speech input users will have difficulty navigating to them.

[Code Reference] 
 ```
<input aria-label="" class="" enterkeyhint="" inputmode="" placeholder="Filter list items" type="text">
 ``` 

### Note: 
 [Suggestion]
Provide a valid label for form fields. Provide aria-label attributes in the ``` <input> ``` elements.

[Compliant Code Example] 
 ```
<input aria-label="Filter list items" class="" enterkeyhint="" inputmode="" placeholder="Filter list items" type="text">
 ```
 

### Media Type: 
 Forms 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by driskull** [comment]
This one would apply to list component as well.

**Comment by geospatialem** [comment]
Updated to apply to the `list` component

**Comment by driskull** [comment]
The fix should be to change from using `aria-label` to `label` within the `calcite-filter` component. 

https://github.com/Esri/calcite-design-system/blob/331247f250ad7bfb892cd77f32c6711c8365bacb/packages/calcite-components/src/components/filter/filter.tsx#L273

In the future, we could add a `filterLabel` property to specifically configure the list filter label.

**Comment by geospatialem** [comment]
For now it seems like its mostly covered in JAWS and NVDA. To @driskull's point above, we could add a new prop and in addition, have a fallback on a built-in translation, something like "Filter list items". Going to up the estimate and move to a future milestone to tackle. cc @brittneytewks

**Comment by geospatialem** [comment]
Verified locally with JAWS and NVDA (refer to context in the PR [here](https://github.com/Esri/calcite-design-system/pull/8069#pullrequestreview-1700750139)). 

Created #8076 for the property for additional context in the future.
